frontend/src/pages/Compare.tsx

```diff
@@ -14,334 +14,44 @@
-            currentList.push({
-              config,
-              runId: runs[i].run!.id!,
-              runName: runs[i].run!.name!,
-            });
-            viewersMap.set(config.type, currentList);
-          });
-        }
-      }),
-    );
-
-    // For each output artifact type, list all artifact instances in all runs
-    this.setStateSafe({ viewersMap });
-  }
-
-  protected _selectionChanged(selectedIds: string[]): void {
-    this.setState({ selectedIds });
-    this._loadParameters(selectedIds);
-    this._loadMetrics(selectedIds);
-  }
-
-  private _collapseAllSections(): void {
-    const collapseSections = {
-      [overviewSectionName]: true,
-      [paramsSectionName]: true,
-      [metricsSectionName]: true,
-    };
-    Array.from(this.state.viewersMap.keys()).forEach(t => {
-      const sectionName = componentMap[t].prototype.getDisplayName();
-      collapseSections[sectionName] = true;
-    });
-    this.setState({ collapseSections });
-  }
-
-  private _loadParameters(selectedIds: string[]): void {
-    const { runs, workflowObjects } = this.state;
-
-    const selectedIndices = selectedIds.map(id => runs.findIndex(r => r.run!.id === id));
-    const filteredRuns = runs.filter((_, i) => selectedIndices.indexOf(i) > -1);
-    const filteredWorkflows = workflowObjects.filter((_, i) => selectedIndices.indexOf(i) > -1);
-
-    const paramsCompareProps = CompareUtils.getParamsCompareProps(filteredRuns, filteredWorkflows);
-
-    this.setState({ paramsCompareProps });
-  }
-
-  private _loadMetrics(selectedIds: string[]): void {
-    const { runs } = this.state;
-
-    const selectedIndices = selectedIds.map(id => runs.findIndex(r => r.run!.id === id));
-    const filteredRuns = runs.filter((_, i) => selectedIndices.indexOf(i) > -1).map(r => r.run!);
-
-    const metricsCompareProps = CompareUtils.multiRunMetricsCompareProps(filteredRuns);
-
-    this.setState({ metricsCompareProps });
+import CompareV1, { CompareState } from './CompareV1';
+import CompareV2 from './CompareV2';
+import { PageProps } from './Page';
+
+export type CompareProps = PageProps & CompareState;
+
+// This is a router to determine whether to show V1 or V2 compare page.
+export default function Compare(props: CompareProps) {
+  const queryParamRunIds = new URLParser(props).get(QUERY_PARAMS.runlist);
+  const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
+
+  // Retrieves run details.
+  const { isSuccess, data } = useQuery<ApiRunDetail[], Error>(
+    ['run_details', { ids: runIds }],
+    () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
+    { staleTime: Infinity },
+  );
+
+  if (data === undefined) {
+    return <></>;
+  }
+
+  // Display the V2 Compare page if all selected runs are V2.
+  if (
+    isFeatureEnabled(FeatureKey.V2_ALPHA) &&
+    isSuccess &&
+    data &&
+    data.every(run => run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest'))
```

Comment:
For the next PR: We should detect the mix of V1 and V2 in the Run List page, so we can prevent the situation where V1 and V2 are mixed here.

Reply:
Ok this sounds good - in the next PR, what should the behavior for the mix of V1 and V2 be? Should the `Compare runs` button be disabled as you mentioned in the project scope? How about if somebody directly types in a URL with a mix of v1 and v2 runs - should it redirect back to the Run List page?

Reply:
As mentioned in another comment: You can use `updateBanner` to tell users about the invalid state in the Compare page: https://github.com/kubeflow/pipelines/blob/48e1e29d9ecd6bef40f40df09709f411d11da365/frontend/src/pages/RunDetails.tsx#L791-L796.  Although it is definitely an option to redirect users back to the Run List page, we lose the opportunity to tell users why we are not showing Compare page in this way.